### PR TITLE
asyncHandler and pagination applied

### PR DESCRIPTION
**Pull request by kennie-larkson**
Features achieved 
Please review and merge or discuss as necessary

Regards

### CODE REVIEW

**Comment by Yusadolat** [comment]
You sent the PR to Master Branch lol

**Comment by Yusadolat** [comment]
You committed only the .ignore file as seen here...

**Comment by kennie-larkson** [comment]
> You sent the PR to Master Branch lol

haba

**Comment by kennie-larkson** [comment]
> You committed only the .ignore file as seen here...

haha
which kain life

**Comment by kennie-larkson** [comment]
> You sent the PR to Master Branch lol

To https://github.com/muslimbuilders/halaqaat-server.git
 * [new branch]      feature/customAsyncHandler-and-pagination-in-halqahController -> feature/customAsyncHandler-and-pagination-in-halqahController
Branch 'feature/customAsyncHandler-and-pagination-in-halqahController' set up to track remote branch 'feature/customAsyncHandler-and-pagination-in-halqahController' from 'origin'.

That what i got on my terminal and I believe that says I created a branch in remote
Or not?

**Comment by kennie-larkson** [comment]
That’s unbelievable though
Check now

**Review by Yusadolat — COMMENTED** [review]
It's updated now

**Comment by Yusadolat** [comment]
The way you name your branch is too long hehehe. It's hard to type and merge from a local computer. 
Another thing is Don't ever send PR to master branch, Send PR to Develop branch Instead

**Comment by kennie-larkson** [comment]
> The way you name your branch is too long hehehe. It's hard to type and merge from a local computer.
> Another thing is Don't ever send PR to master branch, Send PR to Develop branch Instead

Noted
I still dont see how it got to master.
I only created a branch locally and set the same as upstream

File: controllers/halqahController.js [hunk 1]
```diff
@@ -1,64 +1,52 @@
...
-      status: 'success',
-      data: {
-        halqah,
-      },
-    });
-  } catch (err) {
-    console.log(err);
-  }
-};
+  const { params: { id }, body } = req
```
Comment (Yusadolat) [inline controllers/halqahController.js:40]: This can work when I'm changing the whole body. What if I want to change one of those fields.

Comment (kennie-larkson) [reply controllers/halqahController.js:41]: Done 
I have used the PUT verb in the halqahRoute

**Pull request merged**